### ISSUE TITLE
Chrome's default context menu isn't prevented from opening when right-clicking on topics/synapses

### ISSUE THREAD

**Issue by Bortseb** [metamaps/metamaps#658]
I have had this problem for a long while now. When I right click on topics/synapses, both context menus come up. The default chrome one is on top of our custom context menu.

This really gets in my way all the time, and breaks the flow of mapping. (I need to hit esc to remove the default context menu, and see our custom context menu)

I have spent a bit of time trying to solve this issue myself... But now I need some help.

I found this from research:
http://stackoverflow.com/questions/381795/how-to-disable-right-click-context-menu-in-javascript

and I found two places in our code that appear to be attempting to prevent the default context menu from coming up... But its not working it seems (Maybe just for me? Windows 10/Chrome... I haven't heard others mention this problem)

https://github.com/metamaps/metamaps/blob/eef8a281cb53af7ff333f41a6af68241e27c22f2/frontend/src/Metamaps/JIT.js#L1292

https://github.com/metamaps/metamaps/blob/develop/frontend/src/Metamaps/Map/index.js#L42

Maybe @devvmh or @Connoropolous can provide some clues?

**Comment by devvmh** [comment]
I've looked at this before myself, especially after trying to map on Windows. I couldn't figure it out. I can definitely join in taking another look

**Comment by Bortseb** [comment]
Based on this
http://stackoverflow.com/questions/737022/how-do-i-disable-right-click-on-my-web-page

I was trying to find a way to add `oncontextmenu="return false"` onto the canvas element with id infovis-canvas ... but not sure if I did it right to see if this would be a solution.

**Comment by devvmh** [comment]
Yeah i'll bet if you updated that code on L42 it would work. I can't really test this since the deprecated code works, but if you get it working on Windows on a branch, we can just merge it to develop and then verify it's all good on different OSes by using Herokuapp

(heroku gets updated every time develop gets updated)

**Comment by Bortseb** [comment]
What is confusing though is that it only happens when you right click on
topics or synapses... It doesn't happen when you right click on a blank
region of the canvas... Nothing's happens then.

**Comment by devvmh** [comment]
Oh ho, that is interesting. I wonder if we can change the line from 

```
    $('#center-container').bind('contextmenu', function (e) {
      return false
    })
```

to

```
    $('#center-container div').bind('contextmenu', function (e) {
      return false
    })
```

That is, override the contextmenu event on all child divs, not just `#center-container` itself. Or even `$('#center-container *')`, although that might have some unintended consequences too

**Comment by Bortseb** [comment]
All fixed! For reference see #681

**Comment by Bortseb** [comment]
Join Robert on Refind
Just a quick reminder that your invite from Robert Best (chessscholar@gmail.com) is waiting. Don't miss out – come join Refind to discover, save, and read the most relevant stories on the web.

Join now:

https://refind.com/Bortseb?invite=UAsUOQBcTx

-- 
Refind: https://refind.com/
Worth your attention

Unsubscribe: https://refind.com/i/email/unsubscribe/508/invite?address=reply@reply.github.com
